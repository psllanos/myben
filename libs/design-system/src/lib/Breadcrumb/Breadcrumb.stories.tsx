import type { Story, Meta } from '@storybook/react'
import type { BreadcrumbProps } from './Breadcrumb'

import Breadcrumb from './Breadcrumb'

export default {
    title: 'Components/Breadcrumbs',
    component: Breadcrumb,
    parameters: { controls: { exclude: ['as', 'className'] } },
    argTypes: {},
    args: {},
} as Meta

const Template: Story<BreadcrumbProps> = (args) => (
    <Breadcrumb.Group {...args}>
        <Breadcrumb href="/">asdf</Breadcrumb>
        <Breadcrumb>asdf</Breadcrumb>
    </Breadcrumb.Group>
)

export const Base = Template.bind({})
