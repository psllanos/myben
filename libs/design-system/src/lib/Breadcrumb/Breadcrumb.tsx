import type { ReactNode, HTMLAttributes } from 'react'
import { Fragment } from 'react'
import classNames from 'classnames'
import { RiArrowRightSLine } from 'react-icons/ri'
import Link from 'next/link'

export interface BreadcrumbProps {
    href?: string
    className?: string
    children: ReactNode
}

function Breadcrumb({ href, className, children }: BreadcrumbProps): JSX.Element {
    const Tag = href ? 'a' : 'span'
    const inner = <Tag className={classNames('', className)}>{children}</Tag>

    return href ? (
        <Link href={href} legacyBehavior>
            {inner}
        </Link>
    ) : (
        inner
    )
}

export interface BreadcrumbGroupProps extends HTMLAttributes<HTMLDivElement> {
    className?: string
    children: ReactNode[]
}

function Group({ className, children, ...rest }: BreadcrumbGroupProps): JSX.Element {
    return (
        <div
            className={classNames('flex items-center space-x-1 text-gray-100 text-base', className)}
            {...rest}
        >
            {children.map((child, idx) => (
                <Fragment key={`${child}-${idx}`}>
                    <span className={classNames(idx === children.length - 1 && 'text-white')}>
                        {child}
                    </span>
                    {idx < children.length - 1 && <RiArrowRightSLine className="w-4 h-4" />}
                </Fragment>
            ))}
        </div>
    )
}

export default Object.assign(Breadcrumb, { Group })
