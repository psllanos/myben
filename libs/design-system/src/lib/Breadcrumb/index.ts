export { default as Breadcrumb } from './Breadcrumb'
export type { BreadcrumbProps, BreadcrumbGroupProps } from './Breadcrumb'
