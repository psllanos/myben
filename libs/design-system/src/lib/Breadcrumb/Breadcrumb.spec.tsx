import { render, screen } from '@testing-library/react'
import Breadcrumb from './Breadcrumb'

describe('Breadcrumbs', () => {
    describe('when rendered with text and hrefs', () => {
        it('should display the linked text', () => {
            const component = render(
                <Breadcrumb.Group>
                    <Breadcrumb href="/example">Example</Breadcrumb>
                    <Breadcrumb href="/example2">Example 2</Breadcrumb>
                </Breadcrumb.Group>
            )

            expect(screen.getByText('Example')).toBeInTheDocument()
            expect(screen.getByText('Example 2')).toBeInTheDocument()
            expect(component).toMatchSnapshot()
        })
    })
})
