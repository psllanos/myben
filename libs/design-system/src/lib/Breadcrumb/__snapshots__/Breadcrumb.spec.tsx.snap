// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs when rendered with text and hrefs should display the linked text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center space-x-1 text-gray-100 text-base"
      >
        <span
          class=""
        >
          <a
            class=""
            href="/example"
          >
            Example
          </a>
        </span>
        <svg
          class="w-4 h-4"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M13.172 12l-4.95-4.95 1.414-1.414L16 12l-6.364 6.364-1.414-1.414z"
            />
          </g>
        </svg>
        <span
          class="text-white"
        >
          <a
            class=""
            href="/example2"
          >
            Example 2
          </a>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center space-x-1 text-gray-100 text-base"
    >
      <span
        class=""
      >
        <a
          class=""
          href="/example"
        >
          Example
        </a>
      </span>
      <svg
        class="w-4 h-4"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M13.172 12l-4.95-4.95 1.414-1.414L16 12l-6.364 6.364-1.414-1.414z"
          />
        </g>
      </svg>
      <span
        class="text-white"
      >
        <a
          class=""
          href="/example2"
        >
          Example 2
        </a>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
